### PR TITLE
[Connector APIs] Enable Connector APIs framework-wide + update functional tests

### PR DESCRIPTION
**Pull request by jedrazb**
Closes https://github.com/elastic/search-team/issues/8476

### Release Note

Directly accessing connector and sync job state through `.elastic-connectors*` indices is deprecated, and will be disallowed entirely in a future release. Instead, the Elasticsearch Connector APIs should be used. Connectors framework code now uses the Elasticsearch Connector APIs by default.

### Changes

Major change: toggle `feature_use_connectors_api` flag to True

Additionally, use Connector APIs in functional test runner for:
- creating new connector doc
- updating scheduling 
- setting connector configuration 

Cleanup unnecessary boilerplate code for anything other than `configuration` in existing connector.json files. Also format the files with fixed indentation. This was scripted operation so no human errors here (hopefully). 

Update logic where necessary. Update documentation. 

### Bonus

Required Connector APIs endpoints added with newly released ES client 🥳 

### Validation
- Tested locally
- Let's see what buildkite says about this PR

### CODE REVIEW

File: connectors/config.py [hunk 1]
```diff
@@ -86,7 +86,7 @@ def _default_config():
             "initial_backoff_duration": 1,
             "backoff_multiplier": 2,
             "log_level": "info",
-            "feature_use_connectors_api": False,
+            "feature_use_connectors_api": True,
```
Comment (artem-shelkovnikov) [inline connectors/config.py:89]: This should be reflected in the PR - the main goal of the PR is to toggle this flag, tests are side-effect

**Review by artem-shelkovnikov — APPROVED** [review]
Green CI -> Good to go as long as it's manually tested, but I believe you've done basic manual testing?

File: connectors/protocol/connectors.py [hunk 2]
```diff
@@ -172,6 +172,31 @@ async def connector_put(
...
+            scheduling["incremental"] = incremental
+
+        if access_control is not None:
+            scheduling["access_control"] = access_control
+
+        await self.api.connector_update_scheduling(
+            connector_id=connector_id, scheduling=scheduling
+        )
+
+    async def connector_update_configuratiom(
```
Comment (seanstory) [inline connectors/protocol/connectors.py:193]: typo - `configuratiom` vs `configuration`

**Review by seanstory — COMMENTED** [review]
wow this was a lot broader of a change than I'd anticipated. Nice.

Can you add release note to this? I expect it'll supercede https://github.com/elastic/connectors/pull/2884, so would be good to make sure it's release nodes and example config yaml change get covered here.

**Comment by jedrazb** [comment]
I see some proper error with Notion connector (Connector APIs unhappy about its ftest configuration schema) - likely human error when the test was written. Will go over all issues from buildkite and only merge once CI is happy :)

File: connectors/config.py (hunk 1 above)
Comment (navarone-feekery) [reply connectors/config.py:89]: If we're updating the flag in this PR, we should also update the example config files. They default to false.

https://github.com/elastic/connectors/blob/7d82cf09dd9e463a48d028871c6e8dcca86680f6/config.yml.example#L156

Also I'll close https://github.com/elastic/connectors/pull/2884 in favor of this PR.

**Comment by artem-shelkovnikov** [comment]
Me looking at functional tests problems:

image

File: connectors/config.py (hunk 1 above)
Comment (jedrazb) [reply connectors/config.py:89]: > If we're updating the flag in this PR, we should also update the example config files. They default to false.

Ahh yes! I will update this in a followup PR, I missed your comment and don't want to re-trigger the CI (all ftest) to include the proposed change in this PR

**Review by seanstory — APPROVED** [review]

**Comment by seanstory** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[](https://github.com/elastic/connectors/pull/2908)|
|✅|8.16|[](https://github.com/elastic/connectors/pull/2909)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Comment by seanstory** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[](https://github.com/elastic/connectors/pull/2908)|
|✅|8.16|[](https://github.com/elastic/connectors/pull/2909)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Comment by artem-shelkovnikov** [comment]
Should it be 8.17 instead of 8.16?

**Comment by seanstory** [comment]
> Should it be 8.17 instead of 8.16?

Closing the loop, was discussed in https://github.com/elastic/connectors/pull/2909#issuecomment-2426529296

**Pull request merged**